harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -431,12 +429,12 @@ def _train_for_step(self, step_id: int, batches_per_step: int) -> workload.Respo
                     self.context.optimizer.step()
                 self.context.optimizer.zero_grad()
 
-                if self.lr_helper.should_step_lr(
+                if self.context.lr_scheduler.should_step_lr(
                     batches_completed=batch_idx + 1,
                     epoch_length=len(self.training_loader),
                     aggregation_frequency=self.hvd_config.aggregation_frequency,
                 ):
-                    self.lr_helper.step()
+                    self.context.lr_scheduler.step()
```

Comment:
Three questions:
1. should `should_step_lr()` be `_should_call_step()`?
2. should `step()` be `_call_step()`?
3. Why not just put all of this logic into `_call_step()`?

Reply:
Given that the original code in this PR is confusing, I moved the code of the original `_LRHelper` to `PytorchTrial`.